src/jit/hwintrinsicxarch.cpp

```diff
@@ -139,7 +145,15 @@ NamedIntrinsic Compiler::lookupHWIntrinsic(const char* methodName, InstructionSe
                 result = hwIntrinsicInfoArray[i].intrinsicID;
             }
         }
+
+        if ((result > NI_SW_INTRINSIC_START) && (result < NI_HW_INTRINSIC_END))
+        {
+            return NI_Illegal;
+        }
+
+        assert((result > NI_HW_INTRINSIC_START) && (result < NI_SW_INTRINSIC_START));
```

Comment:
This assert will be hit by any function present in ISA static classes and not in one of the tables preventing execution of HW intrinsic APIs  which are not supported by jit.

Reply:
I think the assert should be `result < NI_HW_INTRINSIC_END`

Reply:
With above change it will fir even better.